crates/forge_analyzer/src/definitions.rs

```diff
@@ -2466,6 +2458,23 @@ impl Visit for FunctionCollector<'_> {
                     }
                 }
             }
+            Some(Expr::Object(object_lit)) => {
+                object_lit.props.iter().for_each(|prop| {
+                    if let PropOrSpread::Prop(prop) = prop {
+                        if let Prop::Method(MethodProp { key, function }) = &**prop {
+                            if let Some(key_ident) = key.as_ident() {
+                                self.parent = self.res.get_prop_from_ident(
+                                    self.module,
+                                    id.to_id(),
+                                    key_ident,
+                                );
+                                self.handle_function(&function, None);
+                                self.parent = None;
```

Comment:
Shouldn't we be resetting the parent to the old parent? That way, the parent is more accurately tracked for nested objects. You can save the old parent before the `for_each` call and then restore the parent after it.

Reply:
Fixed. Thank you very much!